### PR TITLE
MRG: Fixes from LGTM

### PR DESCRIPTION
**Pull request by larsoner**
Mostly trivial fixes from `LGTM` except:

1. One in `spectral.py` where I realized there was a bunch of cruft related to `sig_pos_end` that could be eliminated, and the code made more DRY.
2. Modernized `connectivity` tests to utilize `pytest.mark.parametrize`
3. Moved lengthy `spectral_connectivity` explanation to `Notes`.
4. Fixed some XXX's in `mne/gui` to actually probe the `pyface` toolkit being used.
5. Commented out code in `mne/io/bti` that used to live in a `if False:  # XXX : reminds us to support this as we go` block.

### CODE REVIEW

**Comment by larsoner** [comment]
This pull request **introduces 9 alerts** and **fixes 53** when merging a9cb593f638f5616c71b798d1269158ab3b9429b into f7bb2ef4a40bf3bc3e20c82feba145e0e6e35aac - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-6f2d1d47583c998b5ef47235de2fdd2f3304ecf4)

**new alerts:**

* 6 for Signature mismatch in overriding method
* 3 for Variable defined multiple times

**fixed alerts:**

* 37 for Variable defined multiple times
* 6 for Unused local variable
* 3 for Signature mismatch in overriding method
* 3 for Unreachable code
* 1 for Redundant comparison
* 1 for Constant in conditional expression or statement
* 1 for Comparison of identical values
* 1 for Implicit string concatenation in a list

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by larsoner** [comment]
This pull request **fixes 50 alerts** when merging 48c7421a91af81c51b3ab1a29fe386f680a9046d into 57d33722dc80a5065444031068cac94c73aa9568 - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-5197c5ffa0983dd7aec75e854311baec877a1ff8)

**fixed alerts:**

* 37 for Variable defined multiple times
* 6 for Unused local variable
* 3 for Unreachable code
* 1 for Redundant comparison
* 1 for Constant in conditional expression or statement
* 1 for Comparison of identical values
* 1 for Implicit string concatenation in a list

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by larsoner** [comment]
This pull request **fixes 50 alerts** when merging 48c7421a91af81c51b3ab1a29fe386f680a9046d into 57d33722dc80a5065444031068cac94c73aa9568 - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-5197c5ffa0983dd7aec75e854311baec877a1ff8)

**fixed alerts:**

* 37 for Variable defined multiple times
* 6 for Unused local variable
* 3 for Unreachable code
* 1 for Redundant comparison
* 1 for Constant in conditional expression or statement
* 1 for Comparison of identical values
* 1 for Implicit string concatenation in a list

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by larsoner** [comment]
This pull request **fixes 50 alerts** when merging b1b21298f36a9abd6e07ac351b2309d25d7dcede into 57d33722dc80a5065444031068cac94c73aa9568 - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-28fced7ed7b99fd24b2841c20155ea12132dc5e1)

**fixed alerts:**

* 37 for Variable defined multiple times
* 6 for Unused local variable
* 3 for Unreachable code
* 1 for Redundant comparison
* 1 for Constant in conditional expression or statement
* 1 for Comparison of identical values
* 1 for Implicit string concatenation in a list

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: mne/io/bti/bti.py [hunk 1]
```diff
@@ -1292,36 +1290,35 @@ def _get_bti_info(pdf_fname, config_fname, head_shape_fname, rotation_x,
...
-                       rowcals=np.ones(mat.shape[0], dtype='>f4'),
-                       colcals=np.ones(mat.shape[1], dtype='>f4'),
-                       save_calibrated=0)]
-    else:
-        logger.info(
-            'Currently direct inclusion of 4D weight tables is not supported.'
-            ' For critical use cases please take into account the MNE command'
-            ' "mne_create_comp_data" to include weights as printed out by '
-            'the 4D "print_table" routine.')
+    # XXX we should support this someday
```
Comment (massich) [inline mne/io/bti/bti.py]: why don't we remove the whole thing and put it in a PR?

**Comment by massich** [comment]
apart from this commented code, we could merge.

File: mne/io/bti/bti.py (hunk 1 above)
Comment (larsoner) [reply mne/io/bti/bti.py]: Done

https://github.com/mne-tools/mne-python/pull/6024

File: examples/inverse/plot_compute_mne_inverse_epochs_in_label.py [hunk 2]
```diff
@@ -64,11 +64,6 @@
 stcs = apply_inverse_epochs(epochs, inverse_operator, lambda2, method, label,
                             pick_ori="normal", nave=evoked.nave)
 
-stc_evoked = apply_inverse(evoked, inverse_operator, lambda2, method,
-                           pick_ori="normal")
-
-stc_evoked_label = stc_evoked.in_label(label)
```
Comment (agramfort) [inline examples/inverse/plot_compute_mne_inverse_epochs_in_label.py]: Is the in_label present elsewhere in the doc ?

Comment (larsoner) [reply examples/inverse/plot_compute_mne_inverse_epochs_in_label.py]: 11 lines below this

**Comment by larsoner** [comment]
This pull request **fixes 50 alerts** when merging 4632b1a84f3c4c8ee6b9b4fbacc61698015753cb into b757ff19c5556e328111661a304d46dff279d818 - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-0564270d902438f2245dfe351c9f5388db592d11)

**fixed alerts:**

* 37 for Variable defined multiple times
* 6 for Unused local variable
* 3 for Unreachable code
* 1 for Redundant comparison
* 1 for Constant in conditional expression or statement
* 1 for Comparison of identical values
* 1 for Implicit string concatenation in a list

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**